src/js/hyperschedule.js

```diff
@@ -1669,21 +1669,66 @@ function downloadPDF()
           course.courseName,
           width - 12,
         );
+        const courseLocationLines = pdf.splitTextToSize(
+          slot.location,
+          width - 12,
+        );
+        const courseInstructorsLines = pdf.splitTextToSize(
+          course.faculty.map(fullName => fullName.split(",")[0]).join(", "),
+          width - 12
+        );
 
+        // Attributes to be shown are calculated based on space available
+        // order of preference: Code, location, name, professor
+        // order of display: code, name, professor, location
+        const maxTextLength = (yEnd - yStart) - 2 * pdf.getLineHeight();
+        const toLocationLength = courseCodeLines.length +
+              courseLocationLines.length;
+        const toNameLength = toLocationLength + courseNameLines.length;
+        const toInstructorLength = toNameLength + courseInstructorsLines.length;
+        const showLocation = toLocationLength * pdf.getLineHeight() <
+              maxTextLength + .5 * pdf.getLineHeight();
+        const showName = toNameLength  * pdf.getLineHeight() < maxTextLength;
+        const showInstructor = toInstructorLength  * pdf.getLineHeight() < maxTextLength;
+        const totalLength = Math.max(courseCodeLines.length,
+                                    showLocation * toLocationLength,
+                                    showName * toNameLength,
+                                    showInstructor * toInstructorLength);
         const xText = x + width / 2;
-        const yText = (yStart + yEnd) / 2 -
-              (courseCodeLines.length + courseNameLines.length) *
-              pdf.getLineHeight() / 2 +
+        let yText = (yStart + yEnd) / 2 -
+              totalLength * pdf.getLineHeight() / 2 +
               pdf.getLineHeight();
         pdf.setFontStyle("bold");
         pdf.text(xText, yText, courseCodeLines, "center");
+        yText += courseCodeLines.length * pdf.getLineHeight();
         pdf.setFontStyle("normal");
-        pdf.text(
-          xText,
-          yText + courseCodeLines.length * pdf.getLineHeight(),
-          courseNameLines,
-          "center",
-        );
+        if (showName){
+            pdf.text(
+                xText,
+                yText,
+                courseNameLines,
+                "center",
+            );
+            yText += courseNameLines.length * pdf.getLineHeight();
+        }
+        if(showInstructor){
+            pdf.text(
+                xText,
+                yText,
+                courseInstructorsLines,
+                "center",
+            );
+            yText += courseInstructorsLines.length * pdf.getLineHeight();
+        }
+        if(showLocation){
+            pdf.text(
+                xText,
+                yText,
+                courseLocationLines,
+                "center",
+            );
+            yText += courseLocationLines.length * pdf.getLineHeight();
+        }
```

Comment:
I think this code could also be turned into a loop without that much difficulty. It would be a big improvement in my opinion.